### ISSUE TITLE
NEI GUI does not appear when Hackery is installed

### ISSUE THREAD

**Issue by KJ4IPS** [KJ4IPS/Hackery#16]
Reported by Mitchellbrine

"We found a bug with what we believe is Hackery. There is no error or crash, but NEI's GUI will not load if Hackery is also installed in game"

**Comment by KJ4IPS** [comment]
Testing with:
CCC 1.7.10-1.0.4.35
NEI 1.7.10-1.0.4.95

and Hackery built from e31721f4f5496c1d845999feb4ea2bec41eb2b8f

Started game and created new flatland world

BINGO! 

```
[16:47:50] [Client thread/ERROR] [NotEnoughItems]: Error handling SMP Check
java.lang.NullPointerException
	at net.minecraft.potion.PotionHelper.func_77913_a(SourceFile:599) ~[aen.class:?]
	at codechicken.nei.recipe.BrewingRecipeHandler.searchPotions(BrewingRecipeHandler.java:135) ~[BrewingRecipeHandler.class:?]
	at codechicken.nei.api.ItemInfo.searchItems(ItemInfo.java:445) ~[ItemInfo.class:?]
	at codechicken.nei.api.ItemInfo.load(ItemInfo.java:96) ~[ItemInfo.class:?]
	at codechicken.nei.NEIClientConfig.bootNEI(NEIClientConfig.java:258) ~[NEIClientConfig.class:?]
	at codechicken.nei.NEIClientConfig.loadWorld(NEIClientConfig.java:208) ~[NEIClientConfig.class:?]
	at codechicken.nei.NEICPH.handleSMPCheck(NEICPH.java:109) [NEICPH.class:?]
	at codechicken.nei.NEICPH.handlePacket(NEICPH.java:23) [NEICPH.class:?]
	at codechicken.lib.packet.PacketCustom$ClientInboundHandler.handle(PacketCustom.java:98) [PacketCustom$ClientInboundHandler.class:?]
	at codechicken.lib.packet.PacketCustom$CustomInboundHandler.channelRead0(PacketCustom.java:75) [PacketCustom$CustomInboundHandler.class:?]
	at codechicken.lib.packet.PacketCustom$CustomInboundHandler.channelRead0(PacketCustom.java:62) [PacketCustom$CustomInboundHandler.class:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:98) [SimpleChannelInboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) [DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) [DefaultChannelPipeline.class:?]
	at io.netty.channel.embedded.EmbeddedChannel.writeInbound(EmbeddedChannel.java:169) [EmbeddedChannel.class:?]
	at cpw.mods.fml.common.network.internal.FMLProxyPacket.func_148833_a(FMLProxyPacket.java:77) [FMLProxyPacket.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) [ej.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:2049) [bao.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:961) [bao.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:887) [bao.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:148) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.11.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.11.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_25]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_25]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_25]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_25]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:286) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:376) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:165) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:54) [NewLaunch.jar:?]
```

Seems to not like my reflecting to recreate the larger potion array.

**Comment by KJ4IPS** [comment]
Because MC expanded the potion array to 32, we no longer need to reflect it to make it bigger, we can just use one of the free ones (21-31)

**Comment by KJ4IPS** [comment]
Or not... Removing the old potion extension hack did not solve the issue... I think it may be related to NEI's potion search

**Comment by KJ4IPS** [comment]
Darn, this is looking like an NEI bug. However, some changes that were useful were made while investigation, so those were picked into be5e2c8ba07ec23e2155e676c30915491c5de0f4

**Comment by KJ4IPS** [comment]
NEI seems to be dying when it attempts to resolve the effects of a potion item (assuming GlitchPowder)

```java
 int result = PotionHelper.applyIngredient(basePotion, ingred.getItem().getPotionEffect(ingred));
```

**Comment by Mitchellbrine** [comment]
Should this bug be reported to @Chicken-Bones?

**Comment by Laikulo** [comment]
No, it is looking more and more like a side effect of the nasty reflection tricks I pulled to work with the vanilla potion system... Not at home right now, but will dig deeper when I can.

**Comment by Laikulo** [comment]
That part where I realize that the potion is not craft-able... At all... because I did not notice some subtle changes in the potion mappings for 1.7.10. (sigh) [Or not, I just missed some debug code that short-circuited it]

**Comment by Laikulo** [comment]
Looking like it's related to some nasty reflection I pulled to make the potion in hackery even possible. When I next have some time to mess with this, I'll look into what may be going wrong inside PotionHelper...

**Comment by Mitchellbrine** [comment]
Any updates on this?

**Comment by Laikulo** [comment]
I removed the reflection hackery from 1.6. 1.7 extended the potion array, so the evil reflection was not needed. Fixed in 1.7+
